spec/helpers/application_helper/buttons/instance_migrate_spec.rb

```diff
@@ -0,0 +1,31 @@
+describe ApplicationHelper::Button::InstanceMigrate do
+  let(:migration_supported_vm)   { Class.new(VmOrTemplate) { supports(:live_migrate) }.new }
+  let(:migration_unsupported_vm) { Class.new(VmOrTemplate) { supports_not(:live_migrate, :reason => "unavailable") }.new }
```

Comment:
Really interesting!

cc @durandom @blomquisg  ... thought you might like this pluggable testing technique